docs/snakefiles/rules.rst

```diff
@@ -569,6 +569,49 @@ It can for example be used to condition some behavior in the workflow on the exi
         shell:
             "cp {input} {output}"
 
+.. _snakefiles-semantic-helpers-parse-input:
+
+The parse_input function
+"""""""""""""""""""
+
+The ``parse_input`` function allows to parse an input file and return a value.
+It has the signature ``parse_input(input_item, parser, kwargs)``, with ``input_item`` being the key of an input file, ``parse`` being a callable to extract the desired information, and ``kwargs`` extra arguments passed to the parser.
+The function will return the extracted value.
+It can for example be used to extract a value from inside an input file.
+
+.. code-block:: python
+
+    rule a:
+	input:
+	    samples="samples.tsv",
+        output:
+            "samples.id",
+        params:
+            id=parse_input("samples", parser=extract_id)
```

Comment:
See PR https://github.com/snakemake/snakemake/pull/2927, we can now simplify this by just taking a function as first input:

```suggestion
            id=parse_input(input.samples, parser=extract_id)
```

Hence, parse_input should be changed to check whether the first arg is a function, if not it should raise an error pointing to the new docs section in https://github.com/snakemake/snakemake/pull/2927. Parse_input itself would return a function again that internally invokes the other one. I can help with this if it is unclear.